msal/src/main/java/com/microsoft/identity/client/AccountAdapter.java

```diff
@@ -96,10 +118,17 @@ public List<ICacheRecord> filter(@NonNull final List<ICacheRecord> records) {
      */
     private static final CacheRecordFilter guestAccountsWithNoHomeTenantAccountFilter = new CacheRecordFilter() {
 
+        final String TAG = CacheRecordFilter.class.getSimpleName();
+
         private boolean hasNoCorrespondingHomeAccount(@NonNull final ICacheRecord guestRecord,
                                                       @NonNull final List<ICacheRecord> homeRecords) {
+            final String methodTag = TAG + ":hasNoCorrespondingHomeAccount";
             // Init our sought value
             final String guestAccountHomeAccountId = guestRecord.getAccount().getHomeAccountId();
+            if (StringUtil.isNullOrEmpty(guestAccountHomeAccountId)){
+                Logger.warn(methodTag, "Guest account home account id is null or empty.");
```

Comment:
In what cases do this happen?

Reply:
Line 140 shows another case where a NPE can occur, so I'm putting a check here to be precautionary.